projects/core/src/main/java/dan200/computercraft/core/apis/http/NetworkUtils.java

```diff
@@ -150,19 +209,27 @@ public static Options getOptions(String host, InetSocketAddress address) throws
      * @param uri           The URI to connect to.
      * @param socketAddress The address of the socket to connect to.
      * @param sslContext    The SSL context, if present.
+     * @param proxy         The proxy handler, if present.
      * @param timeout       The timeout on this channel.
      * @see io.netty.channel.ChannelInitializer
      */
-    public static void initChannel(SocketChannel ch, URI uri, InetSocketAddress socketAddress, @Nullable SslContext sslContext, int timeout) {
+    public static void initChannel(SocketChannel ch, URI uri, InetSocketAddress socketAddress, @Nullable SslContext sslContext, @Nullable ProxyHandler proxy, int timeout) {
         if (timeout > 0) ch.config().setConnectTimeoutMillis(timeout);
 
         var p = ch.pipeline();
         p.addLast(SHAPING_HANDLER);
 
+        if (proxy != null) {
+            if (CoreConfig.httpProxyType == ProxyType.HTTPS && sslContext != null) {
```

Comment:
`sslContext` will be null when connecting to non-secure/HTTP servers, which is probably not what we want here.

I wonder if we could make `getProxyHandler` return a `Consumer<ChannelPipeline>` instead (where each lambda is just `p -> p.addList(...)`). Then in the HTTPS case we fetch `getSslContext` outside the lambda, and set up the pipeline inside. IDK, I can't think of a cleaner way, but also appreciate it's not very nice.

Reply:
Hopefully I've understood correctly and implemented this properly in 1f5476fa69f2d2689ce828ad28d7c42cd52edd5c. The SSL context is now added to the pipeline inside the lambda in `getProxyHandler`, but it also needs the `SocketChannel` and `timeout` for this.